### PR TITLE
Added a new check for formatted/concatenated strings into template.HTML

### PR DESCRIPTION
**Pull request by minusworld**
Added a new check for formatted/concatenated strings into template.HTML which does not escape input.

### CODE REVIEW

**Comment by minusworld** [comment]
https://dev.massive.ret2.co/triager/filter/716

File: go/net/formatted-template-string.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        return template.HTML("")
+    }
+    return template.HTML(tmpl)
+}
+
+func FormattedInline(r *http.Request) template.HTML {
+    customerId := r.URL.Query().Get("id")
+    // ruleid: formatted-template-string
+    return template.HTML(fmt.Sprintf("<html><body><h1>%s</h1></body></html>", customerId))
+}
```
Comment (nbrahms) [inline go/net/formatted-template-string.go]: Might be worth adding a (presumably currently broken) test case with control flow

File: go/net/formatted-template-string.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func main() {}
```
Comment (nbrahms) [inline go/net/formatted-template-string.go]: This seems like a useful negative test case:

```go
return template.HTML("foo" + "bar")
```

File: go/net/formatted-template-string.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    if (err != nil) {
+        return template.HTML("")
+    }
+    return template.HTML(tmpl)
+}
+
+func FormattedInline(r *http.Request) template.HTML {
+    customerId := r.URL.Query().Get("id")
+    // ruleid: formatted-template-string
+    return template.HTML(fmt.Sprintf("<html><body><h1>%s</h1></body></html>", customerId))
```
Comment (nbrahms) [inline go/net/formatted-template-string.go]: Is there no good way to detect that `customerId` was assigned out of the `Request` object?

Comment (minusworld) [reply go/net/formatted-template-string.go]: In other checks, I used `pattern-inside` to limit checking when `$R *http.Request` was a function argument. I wanted to see what else was out there first: https://dev.massive.ret2.co/triager/filter/716

File: go/net/formatted-template-string.go (hunk 1 above)
Comment (minusworld) [reply go/net/formatted-template-string.go]: Good idea

File: go/net/formatted-template-string.go (hunk 2 above)
Comment (minusworld) [reply go/net/formatted-template-string.go]: Also a good idea

**Pull request merged**